lms/migrations/versions/a012766eb62e_events.py

```diff
@@ -0,0 +1,92 @@
+"""
+events
+
+Revision ID: a012766eb62e
+Revises: b7cbc4eb31d1
+Create Date: 2022-06-27 17:53:12.447500
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "a012766eb62e"
+down_revision = "b7cbc4eb31d1"
+
+
+def upgrade():
+    op.create_table(
+        "lti_event_type",
+        sa.Column("id", sa.Integer(), autoincrement=True, nullable=False),
+        sa.Column(
+            "type",
+            sa.Enum("launch", "deeplinking", name="type", native_enum=False, length=64),
+            nullable=False,
+        ),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk__lti_event_type")),
+    )
+    op.create_table(
+        "lti_event",
+        sa.Column("id", sa.Integer(), autoincrement=True, nullable=False),
+        sa.Column(
+            "created", sa.DateTime(), server_default=sa.text("now()"), nullable=False
+        ),
+        sa.Column("type", sa.Integer(), nullable=True),
+        sa.Column("user_id", sa.Integer(), nullable=True),
+        sa.Column("application_instance_id", sa.Integer(), nullable=False),
+        sa.Column("course_id", sa.Integer(), nullable=False),
+        sa.Column("assignment_id", sa.Integer(), nullable=True),
+        sa.Column(
+            "extra",
+            postgresql.JSONB(astext_type=sa.Text()),
+            server_default=sa.text("'{}'::jsonb"),
+            nullable=False,
+        ),
```

Comment:
This is the one part of the table which has the possibility of tanking performance. We should be pretty restrained about what we put in here.